### ISSUE TITLE
Bearer auth interceptor

### ISSUE THREAD

**Issue by icp1994** [babashka/http-client#49]
Would you be interested in a small contribution for [`:bearer-auth`](https://swagger.io/docs/specification/authentication/bearer-authentication/) similar to the existing `:basic-auth`? It's trivial to use without being in the client anyway but since I'm using it quite a bit thought I would make a PR.

**Comment by borkdude** [comment]
Yeah, I think other projects (clj-http, hato) use `:oauth-token` so let's choose this name instead?

**Comment by icp1994** [comment]
Yes I'd also like the name to be consistent with the ecosystem.
Is the API.md generated with a script or written manually?

**Comment by borkdude** [comment]
API.md is generated with `bb quickdoc`
